src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.CompilationTracker.cs

```diff
@@ -735,6 +744,10 @@ private void RecordAssemblySymbols(Compilation compilation, Dictionary<MetadataR
 
                     RecordSourceOfAssemblySymbol(symbol, projectId);
                 }
+
+                // Record which project this compilation came from.  That way we can easily map back to it when given a
+                // compilation for a compilation-namespace in the future.
+                solutionState._symbolToProjectId.RecordCompilation(compilation, this.ProjectState.Id);
```

Comment:
Is this just working around the earlier TODO in this method? Because then you could just look up in the existing map the compilation's AssemblySymbol?

Reply:
you are 100% correct.  however, due to the problem i'm experiencing with interactive, i wanted to minimize the amount of change i was cuasing (to eliminate possible causes).

Reply:
if we can get interactive working, that's def a change i can/will make in this PR.